atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CassandraRepairHelper.java

```diff
@@ -87,8 +87,6 @@ public void repairInternalTables(Namespace namespace, BiConsumer<String, RangesF
         CqlCluster cqlCluster = cqlClusters.get(namespace);
         KeyedStream.of(getTableNamesToRepair(kvs))
                 .map(tableName -> getRangesToRepair(cqlCluster, namespace, tableName))
-                // TODO(gs): this will do repairs serially, instead of batched. Is this fine? Port batching from
```

Comment:
This is fine for now - context:

I looked into the internal code and chatted with Leon about this - the parallelism logic used internally is quite complex, and we agree that for the purposes of this workstream, it should be fine to keep the parallel side of things in IBP ( (internal backup product).
This means that IBP would call the AtlasBackupService/AtlasRestoreService with one namespace at a time, but would do these calls in parallel.
In practice, the tasks for a single namespace are not done in parallel (for a single namespace, they repair each table in serial, then clean each table in serial) - so it would not harm perf to have IBP call one Atlas restore method that does the repair + clean in serial, provided IBP called Atlas with one namespace at a time. This also makes wiring the Atlas code up to IBP more straightforward.
If we want to further improve performance in future, we could look at moving the parallelism into Atlas and making it more granular, but that’s not needed to satisfy this workstream’s objective, and would add both overhead and risk to do it properly - thus, it’s out of scope.

I’d be happy to add this to an ADR once the IBP wiring is in place, so that we have a more persistent place that we can refer to later, if someone decides to build on it as a hack week project or something